ste/uploader-blobFS.go

```diff
@@ -171,7 +173,8 @@ func (u *blobFSUploader) Epilogue() {
 	if jptm.TransferStatus() > 0 {
 		md5Hash, ok := <-u.md5Channel
 		if ok {
-			_, err := u.fileURL.FlushData(jptm.Context(), jptm.Info().SourceSize, md5Hash)
+			//Type assertions aren't my favorite thing to do but it does work when you need it to.
+			_, err := u.fileURL.FlushData(jptm.Context(), jptm.Info().SourceSize, md5Hash, jptm.(*jobPartTransferMgr).jobPartMgr.(*jobPartMgr).bfsFileHTTPHeaders)
 			if err != nil {
```

Comment:
Looks like this was written before jptm.BfsDstData was added. Now that that function exists, I think this comment can be removed, and the code can be simplified to call jptm.BfsDstData instead of the chain of properties and type assertions. Is that right?

Reply:
*actually* I was trying to do that. I couldn't get it to work, so I resorted to type assertions. I'll take another shot at it.

Reply:
OK, so the reason for that pain is that we can't actually obtain the headers because we don't have access to the leading bytes of the local file.

Believe me, I'd love to make this cleaner. But that means we have to have Epilogue actually get the leading bytes in one way or another, which would end up touching a lot more code. BlobFS is the only one that actually *needs* this info, as FlushData overwrites the headers.

Reply:
Correction: FlushData overwrites x-ms-content-type. If it were anything else, it wouldn't be as concerning since that's actually what the leading bytes manipulate.

Reply:
So, @JohnRusk my question to you is: Should I go through the effort to actually get those leading bytes to jptm.BfsDstData, or should we continue with this one-off hack?

Reply:
Good point. Yes, we should use BfsDstData because otherwise we won't be consistent with the rest of the codebase. Specifically, FileDstData and BlobDst data respect the `NoGuessMimeType` flag, but the current type-assertion approach does not respect that flag (if I understand correctly).

Anyway, good news: you've already got and used the result of `jptm.BfsDstData` in the Prologue method, right?  And that reflects the leading bytes.  So there are two choices for how to handle this code in the epilogue:

1. Test without setting any headers here at all. (I.e. leave the Flush method passing nils for the things that correspond to header contents).  Maybe, as long as _all 5 of them_ are nil, it will not set any of them, and the values set in the prologue will persist. (IIRC that's consistent with how blob storage would approach the same thing - i.e. if you set one you should set them all, but if you set none, that's fine). OR
1. If the above doesn't work, then use the same approach that uploader-azureFiles uses.  It captures headers that that are used in the prologue (when we do have the leading bytes) and uses them again at the end. (uploader-azureFiles does't have the choice of doing option 1, above, due to a different API signature on the method called in the epilogue).

Reply:
Ah, good thought, cache the response. I don't know why that didn't occur to me.